dialer.go

```diff
@@ -120,23 +121,34 @@ func NewDialer(ctx context.Context, opts ...DialerOption) (*Dialer, error) {
 
 // Dial returns a net.Conn connected to the specified Cloud SQL instance. The instance argument must be the
 // instance's connection name, which is in the format "project-name:region:instance-name".
-func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption) (net.Conn, error) {
+func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption) (conn net.Conn, err error) {
+	var endDial trace.EndSpanFunc
+	ctx, endDial = trace.StartSpan(ctx, "cloud.google.com/go/cloudsqlconn.Dial")
```

Comment:
Sorry one comment I forgot to mention - Can we make these specific to an instance? If I have 5 instances, how do I tell which one this is dialing?

Reply:
Definitely. I'll add a tag with the instance. We can annotate the trace with additional information as we like, too.